R/wikidata.R

```diff
@@ -54,44 +55,56 @@ get_wdid <-
       qurl <-
         paste0("wikidata.org/w/api.php?action=wbsearchentities&format=json&type=item")
       qurl <- paste0(qurl, "&language=", language, "&limit=", limit, "&search=", query1)
-      if (verbose)
-        message('Querying ', qurl)
+      if (verbose) webchem_message("query", query, appendLF = FALSE)
       Sys.sleep(0.3)
-      cont <-
-        fromJSON(content(GET(
-          qurl,
-          user_agent('webchem (https://github.com/ropensci/webchem)')
-        ), 'text'))
-      search <- cont$search
-      if (length(search) == 0) {
-        if (verbose)
-          message('Substance not found! Returing NA. \n')
-        id <- NA
-        matched_sub <- NA
-      } else {
-        # use only matches on label
-        search <- search[search$match$type %in% c('label', 'alias'), ]
-        # # check matches
-        search <- search[tolower(iconv(search$match$text,
-                                       "latin1",
-                                       "ASCII",
-                                       sub = "")) == tolower(query), ]
-
-        if (nrow(search) > 1) {
-          id <-
-            matcher(
-              search$id,
-              query = query,
-              result = search$label,
-              match = match,
-              verbose = verbose
-            )
-          matched_sub <- names(id)
+      res <- try(httr::RETRY("GET",
+                             qurl,
+                             httr::user_agent(webchem_url()),
+                             terminate_on = 404,
+                             quiet = TRUE), silent = TRUE)
+      if (inherits(res, "try-error")) {
+        if (verbose) webchem_message("service_down")
+        return(NA)
```

Comment:
errors anyways when service is down at line 112.  Might work if 
```suggestion
      if (inherits(res, "try-error")) {
        if (verbose) webchem_message("service_down")
        return(tibble(query = query, match = NA, wdid = NA)
```